src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,231 @@
+    "PlainExtractor",
+    "SigmaClippingExtractor",
+]
+
+from abc import abstractmethod
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+
+from ctapipe.containers import StatisticsContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    Int,
+    List,
+)
+
+
+class StatisticsExtractor(TelescopeComponent):
+    """Base StatisticsExtractor component"""
+
+    chunk_size = Int(
+        2500,
+        help="Size of the chunk used for the calculation of the statistical values",
+    ).tag(config=True)
+    image_median_cut_outliers = List(
+        [-0.3, 0.3],
+        help="""Interval of accepted image values \\
+                (fraction with respect to camera median value)""",
+    ).tag(config=True)
+    image_std_cut_outliers = List(
+        [-3, 3],
+        help="""Interval (number of std) of accepted image standard deviation \\
+                around camera median value""",
+    ).tag(config=True)
+
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
+        """
+        Base component to handle the extraction of the statistics
+        from a chunk of charges and pulse times (images).
+
+        Parameters
+        ----------
+        kwargs
+        """
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+
+    def __call__(
+        self,
+        dl1_table,
+        masked_pixels_of_sample=None,
+        chunk_shift=None,
+        col_name="image",
+    ) -> list:
+        """
+        Call the relevant functions to extract the statistics
+        for the particular extractor.
+
+        Parameters
+        ----------
+        dl1_table : ndarray
+            dl1 table with images and timestamps stored in a numpy array of shape
+            (n_images, n_channels, n_pix).
+        masked_pixels_of_sample : ndarray
+            boolean array of masked pixels that are not available for processing
+        chunk_shift : int
+            number of samples to shift the extraction chunk
+        col_name : string
+            column name in the dl1 table
+
+        Returns
+        -------
+        List StatisticsContainer:
+            List of extracted statistics and extraction chunks
+        """
+
+        # If no chunk_shift is provided, the chunk_shift is set to self.chunk_size
+        # meaning that the extraction chunks are not overlapping.
+        if chunk_shift is None:
+            chunk_shift = self.chunk_size
+
+        # Function to split table data into appropriated chunks
+        def _get_chunks(col_name):
+            return [
+                (
+                    dl1_table[col_name].data[i : i + self.chunk_size]
+                    if i + self.chunk_size <= len(dl1_table[col_name])
+                    else dl1_table[col_name].data[
+                        len(dl1_table[col_name].data) - self.chunk_size : len(
+                            dl1_table[col_name].data
+                        )
```

Comment:
Do we maybe need a condition somewhere that `len(dl1_table) >= self.chunk_size`? Othwerwise it will silently select a weird chunk here (`len(dl1_table[col_name].data) - self.chunk_size` will get negative)

Reply:
Good point. I think we should catch it at the beginning of this `__call__` and break.